src/test/java/nextstep/subway/station/SectionAcceptanceTest.java

```diff
@@ -0,0 +1,183 @@
+
+        // then
+        List<String> stationNames = response.jsonPath().getList("stations.name", String.class);
+        Long distance = response.jsonPath().getLong("distance");
+        assertAll(
+                () -> assertThat(stationNames).containsOnly(강남역.getName(), 양재시민의숲.getName(), 판교역.getName()),
+                () -> assertThat(stationNames).hasSize(3),
+                () -> assertThat(distance).isEqualTo(lineDistance)
+        );
+    }
+
+    /**
+     * When 노선의 상행 종점을 다음역으로 하는 새로운 역을 생성하면
+     * Then 노선에 새로운 구간이 생성된다
+     * Then 노선의 상행 종점역이 바뀐다
+     */
+    @DisplayName("새로운 역을 상행 종점으로 등록한다.")
+    @Test
+    void addStationInFrontOfUpStation() {
+        // when
+        Long currentDistance = 5L;
+        ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(신분당선.getId(), 신사역.getId(), 강남역.getId(), currentDistance);
+
+        // then
+        List<String> stationNames = response.jsonPath().getList("stations.name", String.class);
+        Long distance = response.jsonPath().getLong("distance");
+        assertAll(
+                () -> assertThat(stationNames).containsOnly(신사역.getName(), 강남역.getName(), 판교역.getName()),
+                () -> assertThat(stationNames).hasSize(3),
+                () -> assertThat(distance).isEqualTo(lineDistance + currentDistance)
+        );
+    }
+
+    /**
+     * When 노선의 하행 종점을 시작점으로 하는 새로운 역을 생성하면
+     * Then 노선에 새로운 구간이 생성된다
+     * Then 노선의 하행 종점역이 바뀐다
+     */
+    @DisplayName("새로운 역을 하행 종점으로 등록한다.")
+    @Test
+    void addStationAfterDownStation() {
+        // when
+        Long currentDistance = 8L;
+        ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(신분당선.getId(), 판교역.getId(), 광교역.getId(), currentDistance);
+
+        // then
+        List<String> stationNames = response.jsonPath().getList("stations.name", String.class);
+        Long distance = response.jsonPath().getLong("distance");
+        assertAll(
+                () -> assertThat(stationNames).containsOnly(강남역.getName(), 판교역.getName(), 광교역.getName()),
+                () -> assertThat(stationNames).hasSize(3),
+                () -> assertThat(distance).isEqualTo(lineDistance + currentDistance)
+        );
+    }
+
+    /**
+     * When 노선에 기존 역 사이 거리보다 크거나 같은 거리의 역을 등록하면
+     * Then 노선에 새로운 구간이 생성이 안된다
+     */
+    @DisplayName("기존 역 사이 거리보다 크거나 같은 거리의 역을 노선에 등록한다.")
+    @ParameterizedTest(name = "기존 역 사이 거리(10)보다 {0}은 크거나 같으므로 실패한다.")
+    @ValueSource(longs = {10L, 12L, 25L})
+    void addSectionWhichHasEqualOrLongerDistance(Long currentDistance) {
+        // when
+        ExtractableResponse<Response> response = 지하철_노선에_지하철역_등록_요청(신분당선.getId(), 강남역.getId(), 양재시민의숲.getId(), currentDistance);
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+    }
+
+    /**
+     * Given 노선에 역들을 생성하고
+     * When 노선에 기존에 등록된 역들의 구간을 등록하면
+     * Then 노선에 새로운 구간이 생성이 안된다
+     */
+    @DisplayName("노선에 기등록된 역들의 구간을 등록한다.")
```

Comment:
주석으로 테스트의 내용을 너무 잘 설명해주셨는데요, `@DisplayName` 에서도 테스트의 결과가 어떻게 되는지 명시되는게 어떨까요?
```suggestion
    @DisplayName("노선에 기등록된 역들의 구간을 등록하면 구간 생성에 실패한다.")
```
위는 예시일 뿐이지만, `@DisplayName` 를 사용하면 전체 테스트를 돌렸을때 목록을 확인할 수 있는데 위처럼 작성하는편이 테스트 코드를 열어보지 않아도 요구사항을 더 쉽게 파악할 수 있을 것 같아요 😁